app/js/skrollr.js

```diff
@@ -25,21 +31,50 @@ angular.module("sn.skrollr", [])
         /**
          * @constructor
          * @param   {Object}  $window    angular wrapper for window
+         * @param   {Object}  $document  angular wrapper for document
          * @param   {Object}  $rootScope angular root application scope
          */
         function($window, $document, $rootScope) {
 
-            return {
+            _this.serviceMethods = {
+
+                /**
+                 * Initialise skrollrjs with config options
+                 * @method init
+                 */
                 init: function() {
 
+                    var skrollrInit = function skrollrInit(){
+                        _this.skrollrInstance = $window.skrollr.init(_this.config);
+                        _this.hasBeenInitialised = true;
+                        _this.serviceMethods.refresh();
+                    };
+
                     $document.ready(function () {
-                        $rootScope.$apply(function() {
-                            var s = $window.skrollr.init(_this.config);
-                        });
+                        if (!$rootScope.$$phase) {
+                            $rootScope.$apply(function() {
+                                skrollrInit();
+                            });
```

Comment:
here you can just pass the variable to the `$rootScope.$apply()` function like so: `$rootScope.$apply(skrollrInit)` rather than wrapping it in an anonymous function